### PR TITLE
CART-822 test: Add checking of DF_RC use to fault injection.

### PR DESCRIPTION
**Pull request by ashleypittman**
Extend fault injection testing to ensure that the DF_RC/DP_RC macros are used
on errors paths that are tested.  This doesn't fix the entire codebase obviously
but defines/enforces what we want messages to look like so when we do extend
the fault injection testing we know what format to set in future.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

**Review by jolivier23 — APPROVED** [review]

**Review by kccain — CHANGES_REQUESTED** [review]

File: src/cart/crt_context.c [hunk 1]
```diff
@@ -293,14 +293,14 @@ crt_rpc_complete(struct crt_rpc_priv *rpc_priv, int rc)
...
 
 		if (cbinfo.cci_rc != 0)
-			RPC_ERROR(rpc_priv, "RPC failed; rc: %d\n",
-				  cbinfo.cci_rc);
+			RPC_ERROR(rpc_priv, "failed, " DF_RC "\n",
+				  DP_RC(cbinfo.cci_rc));
 
 		RPC_TRACE(DB_TRACE, rpc_priv,
-			  "Invoking RPC callback (rank %d tag %d) rc: %d.\n",
+			  "Invoking RPC callback (rank %d tag %d) rc: " DF_RC "\n",
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/cart/crt_context.c:300]: Good catch!  This line (and the matching one in crt_hg.c is used by the log post-processing to see how RPCs are completed, they need to be consistent and need to match the code in cart_logparse.py.

Updated to be consistent, and I've fixed/improved the log analysis as well.

**Review by kccain — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

File: src/cart/crt_context.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (jolivier23) [reply src/cart/crt_context.c:300]: why not just move to next line?   We are very inconsistent in how we handle these issues

[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/cart/crt_context.c:300]: You mean the line length?  Yes, I could do that.

**Comment by ashleypittman** [comment]
I've changed crt_bulk to ensure calls to D_ERROR() don't include the function name, although I'm sure there are many other files with the same issue.

**Comment by ashleypittman** [comment]
Rebased and resolved conflicts.

File: src/gurt/heap.c [hunk 2]
```diff
@@ -497,7 +499,8 @@ d_binheap_insert(struct d_binheap *h, struct d_binheap_node *e)
 	if (h->d_bh_ops->hop_enter) {
 		rc = h->d_bh_ops->hop_enter(h, e);
 		if (rc != 0) {
-			D_ERROR("d_bh_ops->hop_enter failed, rc: %d.\n", rc);
+			D_ERROR("d_bh_ops->hop_enter failed, " DF_RC "\n",
```
Comment (kccain) [inline src/gurt/heap.c:502]: is it worth adding () here?
D_ERROR("d_bh_ops->hop_enter() failed, " DF_RC "\n",

**Review by kccain — APPROVED** [review]
a few questions, approving depending on the answers. looked at the C changes mostly in my review.

File: src/cart/crt_hg.c [hunk 3]
```diff
@@ -177,8 +177,8 @@ crt_hg_pool_init(struct crt_hg_context *hg_ctx)
 	rc = crt_hg_pool_enable(hg_ctx, CRT_HG_POOL_MAX_NUM,
 				CRT_HG_POOL_PREPOST_NUM);
 	if (rc != 0)
-		D_ERROR("crt_hg_pool_enable, hg_ctx %p, failed rc:%d.\n",
-			hg_ctx, rc);
+		D_ERROR("crt_hg_pool_enable, hg_ctx %p, failed rc: " DF_RC "\n",
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/cart/crt_hg.c:180]: Not intentional. there's just a lot to keep in sync, and even now whilst some files may be consistent in the file it's not consistent across the codebase.

File: src/cart/crt_bulk.c [hunk 4]
```diff
@@ -119,7 +118,7 @@ crt_bulk_create(crt_context_t crt_ctx, d_sg_list_t *sgl,
 	ctx = crt_ctx;
 	rc = crt_hg_bulk_create(&ctx->cc_hg_ctx, sgl, bulk_perm, bulk_hdl);
 	if (rc != 0)
-		D_ERROR("crt_hg_bulk_create failed, rc: %d.\n", rc);
+		D_ERROR("crt_hg_bulk_create() failed, rc: %d.\n", rc);
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/cart/crt_bulk.c:121]: Good question, this error was not hit in the fault injection testing which checks for the macro use, but rather was happening on master so it was changed as the result of a different check.

File: src/cart/crt_iv.c [hunk 5]
```diff
@@ -1685,11 +1685,12 @@ crt_iv_fetch(crt_iv_namespace_t ivns, uint32_t class_id,
 
 		if (put_needed)
 			iv_ops->ivo_on_put(ivns, iv_value, user_priv);
-		D_ERROR("Failed to issue IV fetch; rc = %d\n", rc);
+		D_ERROR("Failed to issue IV fetch; rc = " DF_RC "\n",
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/cart/crt_iv.c:1688]: yes, probably.  Will do.

File: src/gurt/heap.c (hunk 2 above)
Comment (ashleypittman) [reply src/gurt/heap.c:502]: This is consistent with the rest of the file, but it's a small file.  Will change.

**Review by kccain — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Comment by ashleypittman** [comment]
Failure is DAOS-6022 I think.

**Pull request merged**